scripts/synthetix/facilicate_trade.js

```diff
@@ -0,0 +1,97 @@
+const { getExchange } = require("../utils/trading_strategy_helpers")(web3, artifacts)
+const { SynthetixJs } = require("synthetix-js")
+const { calculateBuyAndSellAmountsFromPrice, fetchTokenInfoFromExchange } = require("../utils/trading_strategy_helpers")(web3, artifacts)
+
+// TODO - put this somewhere more generic. Copied from dex-contracts
+const sendTxAndGetReturnValue = async function (method, ...args) {
+  const result = await method.call(...args)
+  await method(...args)
+  return result
+}
+
+// truffle uses network by name and SynthetixJS uses network by ID
+const networkMap = {
+  "mainnet": 1,
+  "rinkeby": 4,
+}
+
+const argv = require("yargs")
+  .option("spread", {
+    type: "float",
+    describe: "Percentage increase required for trade (fees not accounted)",
+    default: 0.2,
+  })
+  .version(false).argv
+
+module.exports = async callback => {
+  try {
+    const snxjs = new SynthetixJs({ networkId: networkMap[argv.network] })
```

Comment:
nit: `await web3.eth.net.getId()`

Reply:
Does this really work? I will give it a try. Would be much cleaner.